File: python/python/src/KingPiece.py
'''
Created on Apr 9, 2017

@author: luke
'''

from Piece import Piece
from Coordinate import Coordinate

import MoveRules
    
class KingPiece(Piece):

    def __init__(self, color):
        super().__init__(color, False)
    
    def __repr__(self):
        return "%s King" % self.color
     
    def getSimpleMoves(self, coordinate, board):
        return MoveRules.MoveRules.getSimpleMoves(board, self, coordinate, self.getFollowingCoords)
    
    def getJumpMoves(self, coordinate, board):
        return MoveRules.MoveRules.getJumpMoves(board, self, coordinate, self.getFollowingCoords)
    
    def getFollowingCoords(self, c):
        return [Coordinate(c.row+1, c.col+1), Coordinate(c.row+1, c.col-1), Coordinate(c.row-1, c.col+1), Coordinate(c.row-1, c.col-1)]

File: ruby/KingPiece.rb

require "Piece"

class KingPiece < Piece
	def initialize(color)
		super(color)
		
		@get_following_coords = Proc.new do |c|
			[Coordinate.new(c.row+1, c.col+1), Coordinate.new(c.row+1, c.col-1), 
			Coordinate.new(c.row-1, c.col+1), Coordinate.new(c.row-1, c.col-1)]
		end
	end
		
	def get_jump_moves(coordinate, board)
		MoveRules.get_jump_moves(board, self, coordinate, @get_following_coords)
	end
	
	def get_simple_moves(coordinate, board)
		MoveRules.get_simple_moves(board, self, coordinate, @get_following_coords)
	end
		
	def to_s
		return @color.to_s + " king"
	end
	
	def can_be_kinged?
		return false
	end
end

